Project Finished

HarryjLeigh opened the pull request:
Wrapped up the Drinks Info project and had a great time exploring API calls and presenting the data! Awaiting review :)

Harryjl23 committed:
Project Finished

Harryjl23 committed:
Fix Codacy issues.

Harryjl23 committed:
Fix code formatting

chrisjamiecarter commented on HarryJLeigh.Drinks/DrinksInfo/Controllers/DrinksController.cs:
```diff
@@ -0,0 +1,59 @@
+        }
+    }
+
+    private T? FetchData<T>(string endPoint) where T : class
```
🟢 **Generics**

⭐

chrisjamiecarter commented on HarryJLeigh.Drinks/DrinksInfo/Models/Models.cs:
🟠 **Code Conventions**

💡 Ideally, all classes should be in there own file, in this case it is fine as most are small and you only have one models class in your models folder. But imagine if you have any more classes, you could get to a 100+ line file.

chrisjamiecarter commented on HarryJLeigh.Drinks/DrinksInfo/Program.cs:
🟢 **Program.cs**

⭐  Very lightweight

chrisjamiecarter commented on HarryJLeigh.Drinks/DrinksInfo/README.md:
🟢 **Readme File**

⭐ Great stuff

chrisjamiecarter reviewed (approved):
Hey @HarryjLeigh 👋,

Excellent work on your Drinks project submission 🎉!
I have performed a peer review and have added some praise and suggestions as bespoke comments.
Also, see below for some general feedback.

***

🟠 **Solution Structure**
Minor, but remember it should be `[ProjectName].[GitHubUserName]/` not `[GitHubUserName].[ProjectName]/`

🟠 **Console UI**
Why did you not use Spectre Console for your UI. You could of implemented a selection menu here instead of manual user input

![image](https://github.com/user-attachments/assets/4fe63f04-cd95-4657-b633-e310b374c35d)

🟠 **Lacking Detail**
What about the other information? Do you think the below information allows an employee to create the drink?

![image](https://github.com/user-attachments/assets/609b91b9-2edc-4eeb-9682-a62127e9beed)

***

While not technically against the requirements, I suggest revisiting to re-imagine what data you are returning 🔧 .

Feel free to reach out if you have any questions or if you'd like to collaborate further on any of these points 🆘 .

I will go ahead and mark as approved, keep up the excellent work on the next projects! 😊

Best regards,
@chrisjamiecarter 👍

Harryjl23 committed:
Implemented suggestions
- Spectre console for UI
- Extra Drink detail:
      Added measures column in table next to ingredients
      Added drink instruction after table
- Models separated into individual classes:
      Category.cs
      Drink.cs
      Ingredient.cs
      Instruction.cs
      Measure.cs

Harryjl23 committed:
Project Finished

TheCSharpAcademy commented:
@HarryjLeigh @chrisjamiecarter Has this one been approved in the dashboard?
